src/include/pgmoneta.h

```diff
@@ -180,7 +181,11 @@ struct server
    char backup_slot[MISC_LENGTH];      /**< The backup slot name */
    char wal_slot[MISC_LENGTH];         /**< The WAL slot name */
    char follow[MISC_LENGTH];           /**< Follow a server */
-   int retention;                      /**< The retention for the server */
+   int retention_days;                 /**< The retention days for the server */
+   int retention_weeks;                /**< The retention weeks for the server */
+   int retention_months;               /**< The retention months for the server */
+   int retention_years;                /**< The retention years for the server */
+   int specified_retention_len;        /**< The length of user specified retention parameters*/
```

Comment:
Why do we need this ? Set all variables to -1 (or 0) and work off that

Reply:
Do you mean the specified_retention_len? This is just for convenience, so that we only validate the parameters user specifies, but I guess I can get rid of it

Reply:
Yeah, get rid off `specified_retention_len`

Reply:
Ok I'll work on it. In the meantime could you take a look at the core implementation at `wf_retain.c`? So that I can make changes in one sitting.